drivers/ethernet/eth_stm32_hal_priv.h

```diff
@@ -1,34 +1,104 @@
 /*
  * Copyright (c) 2017 Erwin Rol <erwin@erwinrol.com>
+ * Copyright (c) 2025 STMicroelectronics
  * SPDX-License-Identifier: Apache-2.0
  */
 
 #ifndef ZEPHYR_DRIVERS_ETHERNET_ETH_STM32_HAL_PRIV_H_
 #define ZEPHYR_DRIVERS_ETHERNET_ETH_STM32_HAL_PRIV_H_
 
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
+#include <zephyr/net/phy.h>
 #include <zephyr/kernel.h>
 #include <zephyr/types.h>
+#include <soc.h>
 
-/* Naming of the  ETH PTP Config Status changes depending on the stm32 serie */
-#if defined(CONFIG_SOC_SERIES_STM32F4X)
-#define ETH_STM32_PTP_CONFIGURED HAL_ETH_PTP_CONFIGURATED
-#define ETH_STM32_PTP_NOT_CONFIGURED HAL_ETH_PTP_NOT_CONFIGURATED
+#define DT_DRV_COMPAT st_stm32_ethernet
+
+#define LOG_LEVEL CONFIG_ETHERNET_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
```

Comment:
should only be registered exactly ones the other times it should only be declared

Reply:
So, I'm duplicating all LOG modules declaration to have one per c file, with each its own name. Is that fine for both of you ? @maass-hamburg, @etienne-lms

Reply:
Okay for me:
*eth_stm32_hal_common.c*: `LOG_MODULE_REGISTER(eth_stm32_hal, CONFIG_ETHERNET_LOG_LEVEL);`
*eth_stm32_hal_v1.c*: `LOG_MODULE_DECLARE(eth_stm32_hal);`
*eth_stm32_hal_v2.c*: `LOG_MODULE_DECLARE(eth_stm32_hal);`
*eth_stm32_hal_ptp.c*: `LOG_MODULE_REGISTER(eth_stm32_hal_ptp, CONFIG_ETHERNET_LOG_LEVEL);`